sailfishusermanager.cpp

```diff
@@ -290,11 +291,21 @@ uint SailfishUserManager::addUser(const QString &name)
 
 int SailfishUserManager::removeUserFiles(uint uid)
 {
+    int rv = EXIT_FAILURE;
     QDir dir(USER_ENVIRONMENT_DIR.arg(uid));
     if (dir.removeRecursively())
-        return EXIT_SUCCESS;
-    qCWarning(lcSUM) << "Removing user environment directory failed";
-    return EXIT_FAILURE;
+        rv = EXIT_SUCCESS;
+    else
+        qCWarning(lcSUM) << "Removing user environment directory failed";
+
+    // Execute user removal scripts
+    QDir removal(USER_REMOVE_SCRIPT_DIR);
+    foreach (const QString &entry, removal.entryList(QDir::Files | QDir::Executable)) {
+        if (QProcess::execute(USER_REMOVE_SCRIPT_DIR + '/' + entry, QStringList() << QString::number(uid)))
+            qCWarning(lcSUM) << "User remove script" << USER_REMOVE_SCRIPT_DIR + '/' + entry << "failed";
```

Comment:
It could be useful to return the exit code too.

Reply:
Sorry, not return but print of course!

Reply:
Nit pick: prefer C++ for range over the foreach => also Qt code has been moving away from foreach...

```
for (const QString &entry : removal.entryList(QDir::Files | QDir::Executable)) {
```